tests/brain/test_dataclasses.py

```diff
@@ -1350,3 +1350,23 @@ def attr(self, value: int) -> None:
     fourth_init: bases.UnboundMethod = next(fourth.infer())
     assert [a.name for a in fourth_init.args.args] == ["self", "other_attr", "attr"]
     assert [a.name for a in fourth_init.args.defaults] == ["Uninferable"]
+
+
+def test_dataclass_pylint_8109() -> None:
```

Comment:
Can you move this to the other tests for `format`? This is not really a `dataclass` issue right?

Reply:
Yeah I don't think so but reproducing was not easy with only the format.

Reply:
Well in fact it's (partly) a dataclass issue, the following do not raise anything with the old code:

```python
class Number:
    amount: int | float
    round: int = 2

    def __str__(self): #@
        number_format = "{:,.%sf}" % self.round
        return number_format.format(self.amount).rstrip("0").rstrip(".")
```

Reply:
(I tried to reproduce by adding thing instead of the saner "removing things until it does not crash)

Reply:
Yeah, but it doesn't feel like this should be in the dataclasses test. As in, this is clearly related to our support of `format` in various scenarios instead of our support for `dataclasses`. If I read the `format` tests I would expect this test to be there as well.

Reply:
[b1fd5d7](https://github.com/PyCQA/astroid/pull/2024/commits/b1fd5d71dd3aebd3110d7f67b7c3251ce136dfd8)